src/bin/tui/ui.rs

```diff
@@ -124,6 +133,10 @@ impl UI {
 			return false;
 		}
 
+		while let Some(message) = self.logs_rx.try_iter().next() {
```

Comment:
We may want to rethink some of the tui refresh logic. I think we update the _entire_ tui on every step, regardless of what view or screen in the tui we are actually in.

Particularly for something as heavy as the full screen of log messages, we may want to look into limiting updates here only if we are actually viewing this screen. (Maybe this is a bad idea, I have not thought this through fully).

But it feels like updating many lines of streaming log msg output might be overly expensive if the tui is not actually viewing these at the time. Particularly if the node is logging debug (or trace...) level logs.

Reply:
We do up date the whole thing every second yes. But we could update the logs view only when its the active tab as you suggest. When moving to a separate tab we would clear the logs view and consume but ignore the log messages. So when you went back to the logs view it would start scrolling up from the bottom with the latest messages.

Reply:
Might be worth getting some performance numbers and a rough idea of how costly this is to update before making these changes. Its possible this is not actually an issue and fine for now.

Reply:
So I did some testing and observational measurement between this branch and master and I cannot discern any noticeable impact on resources or responsiveness of the TUI.

Typically grin uses between 10-20% CPU and ~23MB RAM on my machine and this didn't change with TUI trace logging enabled. Also tested with log to file and TUI logging enabled. And tested with chain sync from scratch, again there was no noticeable difference.